pallets/xcm-transactor/src/relay_indices.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+
+#[derive(
+	Clone, Copy, Debug, Default, Deserialize, Serialize, Encode, Decode, TypeInfo, PartialEq, Eq,
+)]
+pub struct RelayChainIndices {
+	// Pallet indices
+	pub staking: u8,
+	pub utility: u8,
+	pub hrmp: u8,
```

Comment:
The names themselves now don't have much context since the nested struct has been removed.

For the pallet ones, renaming them to be e.g. `staking -> pallet_staking` could go a long way.

The others could be prepended with `staking_`, `utility_` and `hrmp_` or something similar to help a bit.

Reply:
...some of the names might get obnoxiously long by doing this, though. I would suggest at least doing the pallet ones for clarity.

Reply:
What would be nice is if `parity_scale_codec` allowed us to express the nested structure but allow us to flatten the whole thing, kind of like `clap`'s `flatten`.

Reply:
That makes sense. Is there anything we can do about it now?

Reply:
Doesn't look like it, I think keeping it flat as you have it now is fine. If anything, we could make a feature request upstream. Parity might like the idea since they encourage shallow decoding schemes for perf reasons...